stream-coordinator/src/main/java/org/apache/kylin/stream/coordinator/Coordinator.java

```diff
@@ -132,14 +143,37 @@ private Coordinator() {
         this.zkClient = ZKUtils.getZookeeperClient();
         this.selector = new CoordinatorLeaderSelector();
         this.jobStatusChecker = new StreamingBuildJobStatusChecker();
-        this.streamingJobCheckExecutor = Executors.newScheduledThreadPool(1, new NamedThreadFactory(
-                "streaming_job_status_checker"));
+        this.streamingJobCheckExecutor = Executors.newScheduledThreadPool(1,
+                new NamedThreadFactory("streaming_job_status_checker"));
         if (ServerMode.SERVER_MODE.canServeStreamingCoordinator()) {
             start();
         }
     }
 
-    public static Coordinator getInstance() {
+    @VisibleForTesting
+    protected Coordinator(StreamMetadataStore metadataStore, ReceiverAdminClient receiverClient) {
+        this.streamMetadataStore = metadataStore;
+        this.receiverAdminClient = receiverClient;
+        this.assigner = new DefaultAssigner();
+        this.zkClient = ZKUtils.getZookeeperClient();
+        this.selector = new CoordinatorLeaderSelector();
+        this.jobStatusChecker = new StreamingBuildJobStatusChecker();
+        this.streamingJobCheckExecutor = Executors.newScheduledThreadPool(1,
+                new NamedThreadFactory("streaming_job_status_checker"));
+        if (ServerMode.SERVER_MODE.canServeStreamingCoordinator()) {
+            start();
+        }
+        this.isLead = true;
+    }
+
+    public static synchronized Coordinator getInstance() {
```

Comment:
late initialization here for unit test?
no need synchronized on method level

Reply:
OK, I understand.